File: app/src/main/java/seriesreminder/ui/menunavigation/MainNavigationModule.java
package seriesreminder.ui.menunavigation;

import dagger.Binds;
import dagger.Module;
import dagger.Provides;

/**
 ** Created by marci on 2018-04-02.
 */
@Module
abstract public class MainNavigationModule {

  @Binds
  abstract MainNavigationContract.View bindView(MainNavigationActivity view);

  @Binds
  abstract MainNavigationContract.Presenter bindPresenter(MainNavigationPresenter presenter);

  @Provides
  static MenuItemParams provideMenuItem(MainNavigationActivity view) {
    return new MenuItemParams(view.getIntent().getExtras());
  }
}

File: app/src/main/java/com/example/marci/seriesreminder/ui/menu_navigation/MainNavigationModule.kt
package com.example.marci.seriesreminder.ui.menu_navigation

import com.example.marcin.meetfriends.ui.menu_navigation.MainNavigationPresenter
import dagger.Binds
import dagger.Module

/**
 * Created by marci on 2017-11-15.
 */
@Module
abstract class MainNavigationModule {

  @Binds
  abstract fun bindView(view: MainNavigationActivity): MainNavigationContract.View

  @Binds
  abstract fun bindPresenter(presenter: MainNavigationPresenter): MainNavigationContract.Presenter
}